src/main/java/com/cloudbees/plugins/credentials/CredentialsDescriptor.java

```diff
@@ -192,4 +257,339 @@ public String getIconClassName() {
+
+                    // what now follows is an example of a realy evil hack, consequently this means...
+                    //
+                    //                         7..       ,
+                    //                      MMM.          MMM.
+                    //                     MMMMM        .MMMMMM
+                    //                    MMMM.           MMMMM.
+                    //                  OMMM                 MMZ
+                    //                  MMM                    MM
+                    //                .MMMM    $.     .       .MM,
+                    //                MMMMM    MMM   MM        MMM
+                    //               .MMMMM. MMMMD  8MMM.     MMMM
+                    //               MMMMMMM.MMMM    MMMMM.  MMMMMM
+                    //               MMMMMMMM.M .     MMM.  MMMMMMM
+                    //               MMMMMMMMM.            MMMMMMMM
+                    //               MMMMMMMMM . ..     MMMMMMMMMMM
+                    //               MMMMMMMM  IMMMM  Z.MMMMMMMMMM ,
+                    //               .MMMMMMM   .M:M   MMMMMMMMMMM M
+                    //              I MMMMMMM.         MMMMMMMMMO  M
+                    //           MMMM  MMMMMMM       .MMMMMMMMM.   .
+                    //         :MMMMMM.MMMMMMM.      MMMMMMMM   .MMMM
+                    //         MMMMMMMMMMMMMMMM      MMMMMMM   MMMMMMMM
+                    //        MMMMMMMMM.MMMMMMM      MMMMMMM MMMMMMMMMM.
+                    //        MMMMMMMMMMMMMMMM?      MMMMMM MMMMMMMMMMM
+                    //        MMMMMMMMMM  .  .       MMMMMIMMMMMMMMMMMM.
+                    //        MMMMMMMMMM               .. :MMMMMMMMMMMM.
+                    //        DMMMMMMMMMM                 MMMMMMMMMMMMM.
+                    //         MMMMMMMMMM.M.              MMMMMMMMMMMM.
+                    //           MMMMMM,                    ....
+                    //
+                    //                   I AM A SAD PANDA
+
+                    List<String> pathSegments = new ArrayList<String>(Arrays.asList(StringUtils.split(path, "/")));
+                    // strip out any leading junk
+                    while (!pathSegments.isEmpty() && StringUtils.isBlank(pathSegments.get(0))) {
+                        pathSegments.remove(0);
+                    }
+                    if (pathSegments.size() >= 2) {
+                        String firstSegment = pathSegments.get(0);
+                        if ("user".equals(firstSegment)) {
+                            User user = User.get(pathSegments.get(1));
+                            if (type.isInstance(user) && CredentialsProvider.hasStores(user)) {
+                                // we have a winner
+                                return type.cast(user);
+                            }
+                        } else if ("job".equals(firstSegment) || "item".equals(firstSegment) || "view"
```

Comment:
Yes, I didn't write mine myself and IIRC it doesn't work well with views and folders :) But hardly none has complained, probably because it isn't a well known feature.